geoAssembler/widgets/qt_objects.py

```diff
@@ -158,36 +158,45 @@ def _apply(self):
         self.filename_set_signal.emit()
         self.close()
 
-class QLogger(logging.Handler):
-    """Logger object connected python logging."""
 
-    def __init__(self, main_widget):
-        """Create a Dialog that displays the log connected to logging.
+class LogCapturer(logging.Handler, QtCore.QObject):
+    """Emit Qt signal for Python logging, and store recent messages.
+    """
+    new_msg = QtCore.Signal(str)
 
-        Parameters:
-            main_widget : Parent creating this dialog
-        """
-        super().__init__()
-        self.win = QtGui.QDialog(main_widget)
-        layout = QtWidgets.QGridLayout()
-
-        self.widget = QtGui.QPlainTextEdit()
-        self.widget.setReadOnly(True)
-        layout.addWidget(self.widget, 0, 0, 10, 10)
-        ok_btn = create_button('Ok', 'ok')
-        ok_btn.clicked.connect(lambda: self.win.close())
-        layout.addWidget(ok_btn, 11, 0, 1, 1)
-        self.win.setLayout(layout)
-
-    def show(self):
-        """Show the log window."""
-        self.win.show()
+    def __init__(self, parent, level=logging.NOTSET):
+        QtCore.QObject.__init__(self, parent)
+        logging.Handler.__init__(self, level)
+        self.msg_buffer = deque(maxlen=500)
 
     def emit(self, record):
-        """Overload emit signal to write into text widget."""
         msg = self.format(record)
-        self.widget.appendPlainText(msg)
+        self.msg_buffer.append(msg)
+        self.new_msg.emit(msg)
+
+
+class LogDialog(QtGui.QDialog):
+    """A Dialog that displays the log connected to logging.
+
+    Parameters:
+        main_widget : Parent creating this dialog
+    """
+    def __init__(self, main_window):
+        super().__init__(parent=main_window)
+        self.setAttribute(QtCore.Qt.WA_DeleteOnClose)
+        self.log_capturer = main_window.log_capturer
+
```

Comment:
you can also add a window title here like: `self.setWindowTitle('GeoAssembler Logs:')`

Reply:
Good idea, will do